app/models/pageflow/membership.rb

```diff
@@ -1,9 +1,26 @@
 module Pageflow
   class Membership < ActiveRecord::Base
     belongs_to :user
-    belongs_to :entry
+    belongs_to :entity, polymorphic: true
 
-    validates :user, :entry, :presence => true
-    validates :entry_id, :uniqueness => { :scope => :user_id }
+    validates :user, :entity, :role, presence: true
+    validates :user_id, uniqueness: {scope: [:entity_type, :entity_id]}
+    if :entity_type == 'Pageflow::Entry'
```

Comment:
This will always be false. We will have to use `if` options for the `validates` call.